conda/cli/main_search.py

```diff
@@ -25,6 +27,29 @@ def execute(args, parser):
     else:
         subdirs = context.subdirs
 
+    if args.envs:
+        with spinner("Searching environments for %s" % spec,
+                     not context.verbosity and not context.quiet,
+                     context.json):
+            prefix_matches = search_all_prefixes(spec)
+        formatted_result = tuple({
+            'location': prefix,
+            'package_refs': tuple(PackageRef.from_objects(prefix_rec)
+                                  for prefix_rec in prefix_recs),
+        } for prefix, prefix_recs in iteritems(prefix_matches))
+        if context.json:
+            stdout_json(formatted_result)
+        else:
+            builder = []
+            for pkg_group in formatted_result:
+                builder.append("location: %s" % pkg_group['location'])
+                builder.append("package matches:%s" % dashlist(
+                    pref.dist_str() for pref in pkg_group['package_refs']
+                ))
+                builder.append('')
+            print('\n'.join(builder))
```

Comment:
That'd be the third `'\n'.join(builder)` in this file -- time for a little refactoring 😉?

Reply:
Probably, but PRs are hard enough to get merged as it is right now, without making them *look* much bigger than they actually by adding in a bunch of refactored code.  If we're able to someday get more resources on conda to merge PRs, then definitely.